ssv/value_check.go

```diff
@@ -36,45 +37,43 @@ func dutyValueCheck(
 	return nil
 }
 
-func AttesterValueCheckF(
+func BeaconVoteValueCheckF(
 	signer types.BeaconSigner,
-	network types.BeaconNetwork,
-	validatorPK types.ValidatorPK,
-	validatorIndex phase0.ValidatorIndex,
-	sharePublicKey []byte,
+	slot phase0.Slot,
+	sharePublicKeys []types.ShareValidatorPK,
+	estimatedCurrentEpoch phase0.Epoch,
 ) qbft.ProposedValueCheckF {
 	return func(data []byte) error {
-		cd := &types.ConsensusData{}
-		if err := cd.Decode(data); err != nil {
-			return errors.Wrap(err, "failed decoding consensus data")
-		}
-		if err := cd.Validate(); err != nil {
-			return errors.Wrap(err, "invalid value")
+		bv := types.BeaconVote{}
+		if err := bv.Decode(data); err != nil {
+			return errors.Wrap(err, "failed decoding beacon vote")
 		}
 
-		if err := dutyValueCheck(&cd.Duty, network, types.BNRoleAttester, validatorPK, validatorIndex); err != nil {
-			return errors.Wrap(err, "duty invalid")
-		}
-
-		attestationData, _ := cd.GetAttestationData() // error checked in cd.validate()
-
-		if cd.Duty.Slot != attestationData.Slot {
-			return errors.New("attestation data slot != duty slot")
+		if bv.Target.Epoch > estimatedCurrentEpoch+1 {
+			return errors.New("attestation data target epoch is into far future")
 		}
 
-		if cd.Duty.CommitteeIndex != attestationData.Index {
-			return errors.New("attestation data CommitteeIndex != duty CommitteeIndex")
+		if bv.Source.Epoch >= bv.Target.Epoch {
+			return errors.New("attestation data source >= target")
 		}
 
-		if attestationData.Target.Epoch > network.EstimatedCurrentEpoch()+1 {
-			return errors.New("attestation data target epoch is into far future")
+		attestationData := &phase0.AttestationData{
```

Comment:
not sure what you mean

Reply:
The difference between AttestationData and BeaconVote lies in CommitteeIndex.
We don't come to consensus on this field since different validators may be on different committees..

Eth slashing check doesn't take this field into account